sstables/sstables.cc

```diff
@@ -2369,7 +2369,7 @@ static entry_descriptor make_entry_descriptor(sstring sstdir, sstring fname, sst
     } else {
         throw malformed_sstable_exception(seastar::format("invalid version for file {}. Name doesn't match any known version.", fname));
     }
-    return entry_descriptor(sstdir, ks, cf, generation_from_value(boost::lexical_cast<unsigned long>(generation)), version, sstable::format_from_sstring(format), sstable::component_from_sstring(version, component));
+    return entry_descriptor(sstdir, ks, cf, generation_from_value(boost::lexical_cast<int64_t>(generation)), version, sstable::format_from_sstring(format), sstable::component_from_sstring(version, component));
```

Comment:
can be submitted as a standlalone patch (even a standalone PR so it can be merged earlier)